src/lightning_lite/lite.py

```diff
@@ -219,8 +219,9 @@ def setup_module(self, module: nn.Module, move_to_device: bool = True) -> _LiteM
         module = self._strategy.setup_module(module)
         module = _LiteModule(module, self._precision, original_module=original_module)
 
-        # Update the _DeviceDtypeModuleMixin's device parameter
-        module.to(self.device if move_to_device else next(module.parameters()).device)
+        if not isinstance(self._strategy, FSDPStrategy):
+            # Update the _DeviceDtypeModuleMixin's device parameter
```

Comment:
Should we still set the `_DeviceDtypeModuleMixin` device even though we don't move the module itself? It's a bit unclear what's the expectation for the device with shared models.

Reply:
> Should we still set the _DeviceDtypeModuleMixin device even though we don't move the module itself?

That's exactly what I'm avoiding here. The self.device will have an incorrect value anyway for sharded models. FYI the line of code here is only supposed to handle a special case in which you did
```
model.to(...)
self.setup(model, move_to_device=False)
```
in which case we needed to call .to() again internally to update the internal self.device on the wrapper to the correct value.

> It's a bit unclear what's the expectation for the device with shared models.

Yes. I think that was a concern for a while already. User feedback will tell us whether we should drop it, redesign it, or keep it.